src/mon/NVMeofGwMon.cc

```diff
@@ -329,8 +324,9 @@ bool NVMeofGwMon::preprocess_command(MonOpRequestRef op)
     if (HAVE_FEATURE(mon.get_quorum_con_features(), NVMEOFHA)) {
       f->dump_string("features", "LB");
       if (map.created_gws[group_key].size()) {
-        uint32_t index = (global_rebalance_index %
-              map.created_gws[group_key].size()) + 1;
+        time_t seconds_since_1970 = time(NULL);
+        uint32_t index = ((seconds_since_1970/60) %
```

Comment:
I was summoned to look at an unrelated check failure and have zero context (and an empty commit message is of no help), but this seems suspicious to me.  The index (`rebalance_ana_group`) can still change between two successive `nvme-gw show` command invocations, only instead of changing every 5 seconds it's now going to change every 60 seconds.  Is that intended?  I'd have expected the index to be derived from the map epoch, so that it changes only when a new map is published.

Reply:
@idryomov , the requirement   for this feature was that all nvmeofGW-monitors report the same number when nvme-gw show command is invoked. The same published number lasts for 1 minute. Any monitor(leader or peons ) can answer this command. For this purpose I don't need any data related to the epoch. I need some reference. the time(NULL) gives me this reference. I divide it on 60 (complete division) to get the current minute.

Reply:
> the requirement for this feature was that all nvmeofGW-monitors report the same number when nvme-gw show command is invoked.

How often `nvme-gw show` command gets invoked?  Can it be invoked twice in short succession or even just twice (or more) within any 60 second interval?  If so, different numbers would be reported even by the same monitor, let alone different monitors.

Reply:
The purpose of this calculation is only to define which GW will drive the load balancing process in the next 60 seconds. So long that all of the GWs when they run "gw show" get the same index, its ok. In the next 60 seconds, another GW will get the reasonability to drive the LB process, and so on. 
@idryomov why do you think that if the command is invoked few times in the same 60 seconds slot, the result will be different?

Reply:
> why do you think that if the command is invoked few times in the same 60 seconds slot, the result will be different?

We need to define "any 60 second interval" (my comment) and "same 60 second slot" (your comment) to get the bottom of the requirement here.  I meant any interval, starting at an arbitrary `seconds_since_1970`.  Did you mean the same or rather 60 second "steps" (i.e. 0..59, 60..119, etc)?

Let's say `nvme-gw show` gets invoked twice in short succession, by gateway A at `seconds_since_1970` 8999 and gateway B at `seconds_since_1970` 9000.  With 4 gateways, gateway A would get index 2 while gateway B would get index 3, despite the invocations being just a millisecond away from each other.  Is that OK?

>  the requirement for this feature was that all nvmeofGW-monitors report the same number when nvme-gw show command is invoked [...] Any monitor(leader or peons ) can answer this command.

Was a possibility of a clock skew between monitors considered?

Reply:
@idryomov , in the usecase you present  nothing bad  happen .  In some cases 2 GWs would perform re-balancing in the same time - during short interval. We thought about this scenario when we did design.  No harm would be done but in some cases these 2 GWs may choose the same destination ANA group  and it would be eventually overloaded by one (also not always).  this is a multiplication of probabilities   (2GW works together)*(they chose the same destination ANA)*(this destination ANA would be overloaded) . Even in this case another re-balance action would fix this